### PR TITLE
segas32.cpp: Add save states, Fix backdrop color fill when screen resolution is changed

### PR DESCRIPTION
**Pull request by cam900**
Add notes in segas32.h

### CODE REVIEW

File: src/mame/includes/segas32.h [hunk 1]
```diff
@@ -256,6 +261,9 @@ class segas32_state : public device_t
 	uint8_t m_sprite_control_latched[8];
 	uint8_t m_sprite_control[8];
 	std::unique_ptr<uint32_t[]> m_spriteram_32bit;
+	std::unique_ptr<int[]> m_prev_startx;
+	std::unique_ptr<int[]> m_prev_endx;
+	std::unique_ptr<int[]> m_bgcolor_line;
```
Comment (cuavas) [inline src/mame/includes/segas32.h:266]: You need to use explicit-sized types for things that you’re saving for portability.

File: src/mame/video/segas32.cpp [hunk 2]
```diff
@@ -268,22 +268,51 @@ void segas32_state::device_start()
...
+
+	/* save states */
+	save_item(NAME(m_v60_irq_control));
+	save_item(NAME(m_sound_irq_control));
+	save_item(NAME(m_sound_irq_input));
+	save_item(NAME(m_sound_dummy_value));
+	save_item(NAME(m_sound_bank));
+
+	for (int i = 0; i < 2; i++)
+		save_item(NAME(m_mixer_control[i]), i);
```
Comment (cuavas) [inline src/mame/video/segas32.cpp:305]: This shouldn’t need a loop - MAME can save multi-dimensional arrays.

**Review by cuavas — CHANGES_REQUESTED** [review]

**Comment by angelosa** [comment]
Again, what is supposed to fix? 
From my mass testing I don't remember such thing as dynamic screen resolution changes that affects back layer. 
Plus iirc only two games even uses the feature, Arabian Fight and Kokoroji 2 ...

**Comment by cam900** [comment]
This also fixes darkedge warning screen. (sometimes cropped right)

**Pull request merged**